### PR TITLE
Optimize to judge whether it is a power of two

### PR DESCRIPTION
**Pull request by skyguard1**
Motivation:

Optimize to judge whether it is a power of two.please review the pr,thk.

Modification:

Result:

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by skyguard1** [comment]
the stack trace is:
io.netty.handler.codec.CorruptedFrameException: invalid JSON received at byte position 24: 4573636170656420646f755c22626c652071756f746573205c2220696e204a534f4e207374725c22696e6722
	at io.netty.handler.codec.json.JsonObjectDecoder.decode(JsonObjectDecoder.java:174)

Maybe I forgot to consider some aspects, so I'm going to make some changes to run this test case

**Comment by skyguard1** [comment]
I made a stupid mistake. . . , 1 is the power of 2, but here 1 should return false

File: codec/src/main/java/io/netty/handler/codec/json/JsonObjectDecoder.java [hunk 1]
```diff
@@ -213,7 +213,7 @@ private void decodeByte(byte c, ByteBuf in, int idx) {
                     }
                 }
                 // The double quote isn't escaped only if there are even "\"s.
-                if (backslashCount % 2 == 0) {
```
Comment (franz1981) [inline codec/src/main/java/io/netty/handler/codec/json/JsonObjectDecoder.java:216]: Have you looked what the JIT produce here?

Comment (skyguard1) [reply codec/src/main/java/io/netty/handler/codec/json/JsonObjectDecoder.java:216]: Yes, jit is optimized. . . Thanks for pointing out

**Comment by skyguard1** [comment]
So this is a wrong pr...can turn it off, sorry

**Pull request closed without merging**